documentation/docs/extensibility.md

```diff
@@ -1,86 +1,214 @@
-  }
-  return this
-  ```
+Here is a simple example for such an extension, which you can use as a starting point:
+
+```groovy
+void call(Map params) {
+  //access stage name
+  echo "Start - Extension for stage: ${params.stageName}"
+
+  //access config
+  echo "Current stage config: ${params.config}"
+
+  //execute original stage as defined in the template
+  params.originalStage()
+
+  //access overall pipeline script object
+  echo "Branch: ${params.script.commonPipelineEnvironment.gitBranch}"
+
+  echo "End - Extension for stage: ${params.stageName}"
+}
+return this
+```
+
+!!! note "`return this`"
+    Don't forget the `return this`, which is required at the end of _all_ extension scripts.
+    This is due to how Groovy loads scripts internally.
 
 !!! note "Init stage cannot be extended"
-    Please note, the `Init` stage among other things also checks out your current repository.<br />Thus it is not possible to use extensions on this stage.
+    Please note that the `Init` stage also checks out your current repository including your extensions.<br />
+    Therefore, it is not possible to use extensions on this stage.
+
+### Practical example
+
+For a more practical example, you can use extensions in the SAP Cloud SDK pipeline to add custom linters to your pipeline.
+
+A linter is a tool that can check the source code for certain stylistic criteria. Many teams choose to use a linter to ensure a common programming style.
+
+For example, if you want to use [Checkstyle](https://checkstyle.sourceforge.io/) in your codebase, you might use an extension similar to this one in a file called `.pipeline/extensions/lint.groovy` in your project:
+
+```groovy
+def call(Map parameters) {
+
+    parameters.originalStage() // Runs the built in linters
+
+    mavenExecute(
+        script: parameters.script,
+        flags: '--batch-mode',
+        pomPath: 'application/pom.xml',
+        m2Path: s4SdkGlobals.m2Directory,
+        goals: 'checkstyle:checkstyle',
+    )
+
+    recordIssues blameDisabled: true,
+        enabledForFailure: true,
+        aggregatingResults: false,
+        tool: checkStyle()
+}
+
+return this
+```
+
+This code snippet has three components, let's see what is happening here:
+
+Firstly, we run the original stage.
+This runs the SAP UI5 Best Practices linter as this is a standard feature of SAP Cloud SDK pipeline.
+
+Secondly, we run the checkstyle maven plugin using the `mavenExecute` Jenkins library step as provided by project "Piper".
+This serves as an example for how flexible you can re-use what project "Piper" already provides in your extension.
+
+Finally, we use the Jenkins [Warnings NG plugin](https://plugins.jenkins.io/warnings-ng/) and its step `recordIssues` to make the findings visible in the Jenkins user interface.
+
+This example can be adapted for other linters of your choice.
+Be sure to checkout the _Library steps_ section of this documentation if you want to do this.
```

Comment:
I misunderstood! You are referring to the section in the table of contents. I don't know how to write this better... Maybe something in the line of: "Be sure to checkout the specific step you want to adapt in the _Library steps_ section of this documentation..."